src/python/WMCore/FwkJobReport/Report.py

```diff
@@ -599,16 +601,11 @@ def addError(self, stepName, exitCode, errorType, errorDetails, siteName=None):
         errDetails.type = str(errorType)
 
         try:
-            if hasattr(errorDetails, "decode"):
-                # Fix for the unicode encoding issue, #8043
-                # interprets this string using utf-8 codec and ignoring any errors
-                errDetails.details = errorDetails.decode('utf-8', 'ignore')
-            else:
-                # Then cast it to string and decode it
-                errorDetails = str(errorDetails)
-                errDetails.details = errorDetails.decode('utf-8', 'ignore')
+            errDetails.details = newstr(errorDetails)
         except UnicodeEncodeError as ex:
             msg = "Failed to decode the job error details for job ID: %s." % self.getJobID()
+            # if the `try` fails in py3, then the following line will also
+            # fail in python3 and there is not much that we can do
```

Comment:
Alan, we can remove this comment if you think it is irrelevant!

Reply:
Is it because it will fail to stringify `errorDetails` in the error message?